File: internal/domain/menu.go
package domain

import (
	"context"
	"time"
)

type Menu struct {
    ID           string
    RestaurantID string
    Version      int
    IsPublished  bool
    PublishedAt  time.Time
    Tabs         []Tab
    CreatedAt    time.Time
    UpdatedAt    time.Time
    UpdatedBy    string
    IsDeleted    bool
    ViewCount    int
}

type Tab struct {
    ID         string
    MenuID     string
    Name       string
    NameAm     string
    Categories []Category
    IsDeleted  bool
}

type Category struct {
    ID     string
    TabID  string
    Name   string
    NameAm string
    Items  []Item
}



type IMenuUseCase interface {
    CreateMenu(menu *Menu) error
    UpdateMenu(id string, menu *Menu) error
    PublishMenu(id string, userID string) error
    GetMenuByID(id string) (*Menu, error)
    GenerateQRCode(menuID string) (*QRCode, error)
    DeleteMenu(id string) error
}

type IMenuRepository interface {
    Create(ctx context.Context, menu *Menu) error
    Update(ctx context.Context, id string, menu *Menu) error
    GetByID(ctx context.Context, id string) (*Menu, error)
    Delete(ctx context.Context, id string) error
}

File: dinq-mobile/dinq/lib/features/restaurant_management/domain/entities/menu.dart
import 'package:equatable/equatable.dart';

import 'tab.dart';

class Menu extends Equatable {
  final String id;
  final String restaurantId;
  final bool isPublished;
  final List<Tab> tabs;
  final int viewCount;

  const Menu({
    required this.id,
    required this.restaurantId,
    required this.isPublished,
    required this.tabs,
    
    required this.viewCount,
  });

  @override
  List<Object?> get props {
    return [id];
  }
}
